tests/backends/mysql/test_operations.py

```diff
@@ -0,0 +1,98 @@
+import unittest
+
+from django.core.management.color import no_style
+from django.db import connection
+from django.test import SimpleTestCase
+
+from ..models import Person, Tag
+
+
+@unittest.skipUnless(connection.vendor == 'mysql', 'MySQL tests')
+class MySQLOperationsTests(SimpleTestCase):
+    def test_sql_flush(self):
+        statements = connection.ops.sql_flush(
+            no_style(),
+            [Person._meta.db_table, Tag._meta.db_table],
+            [],
+        )
+        self.assertEqual(statements[0], 'SET FOREIGN_KEY_CHECKS = 0;')
+        # The tables are processed in an unordered set.
+        self.assertEqual(
+            sorted(statements[1:-1]),
+            [
+                'DELETE FROM `backends_person`;',
+                'DELETE FROM `backends_tag`;',
+            ],
+        )
+        self.assertEqual(statements[-1], 'SET FOREIGN_KEY_CHECKS = 1;')
+
+    def test_sql_flush_sequences(self):
+        statements = connection.ops.sql_flush(
+            no_style(),
+            [Person._meta.db_table, Tag._meta.db_table],
+            [
+                {
+                    'table': Person._meta.db_table,
+                    'column': Person._meta.pk.db_column,
+                },
+                {
+                    'table': Tag._meta.db_table,
+                    'column': Tag._meta.pk.db_column,
+                },
+            ],
+        )
+        self.assertEqual(statements[0], 'SET FOREIGN_KEY_CHECKS = 0;')
+        # The tables are processed in an unordered set.
+        self.assertEqual(
+            sorted(statements[1:-1]),
+            [
+                'TRUNCATE `backends_person`;',
+                'TRUNCATE `backends_tag`;',
+            ],
+        )
+        self.assertEqual(statements[-1], 'SET FOREIGN_KEY_CHECKS = 1;')
+
+    def test_sql_flush_allow_cascade(self):
+        statements = connection.ops.sql_flush(
+            no_style(),
+            [Person._meta.db_table, Tag._meta.db_table],
+            [],
+            allow_cascade=True,
+        )
+        self.assertEqual(statements[0], 'SET FOREIGN_KEY_CHECKS = 0;')
+        # The tables are processed in an unordered set.
+        self.assertEqual(
+            sorted(statements[1:-1]),
+            [
+                'DELETE FROM `backends_person`;',
+                'DELETE FROM `backends_tag`;',
+            ],
+        )
+        self.assertEqual(statements[-1], 'SET FOREIGN_KEY_CHECKS = 1;')
```

Comment:
`allow_cascade` doesn't change anything on MySQL, so I'll merge this test with `test_sql_flush()`.

Reply:
That's fine with me -- I don't use MySQL -- but that was the point to include it. To show:

1. `allow_cascade` has no effect as one could compare it with the other tests.
2. Should the code change so `allow_cascade` does have an effect, it will be immediately obvious through existing tests.

Reply:
Ah, I misunderstood. I now see that both `allow_cascade=True` and `allow_cascade=False` are executed by the other tests. Sorry for the confusion. That makes sense to me.